presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentControlUtils.java

```diff
@@ -0,0 +1,220 @@
+            return Response.status(NOT_FOUND).build();
+        }
+        try {
+            int prestoRestart = executeCommand("service", "presto", "restart");
+            if (prestoRestart != 0) {
+                LOGGER.error("Failed to restart presto; Process exited with return value: {}", prestoRestart);
+            }
+        }
+        catch (PrestoManagerException  e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    public static Response prestoStatus()
+    {
+        if (!isInstalled()) {
+            return Response.status(OK).entity(new Gson().toJson("Presto is not installed"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+        if (!isRunning()) {
+            return Response.status(OK).entity(new Gson().toJson("Presto is installed but not running"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+        try {
+            String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+            return newRequestBuilder(new URI(format("localhost:%s/v1/info", prestoPort))).buildGet().invoke();
+        }
+        catch (IOException | URISyntaxException e) {
+            LOGGER.error("Failed to get status", e);
+            return Response.status(INTERNAL_SERVER_ERROR).entity(new Gson().toJson("Failed to get status"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+    }
+
+    private static boolean isInstalled()
+    {
+        try {
+            if (executeCommand("rpm", "-q", "presto-server-rpm") != 0) {
+                LOGGER.error("Presto is not installed");
+                return false;
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+        }
+        return true;
+    }
+
+    private static boolean isRunning()
+    {
+        try {
+            if (executeCommand("service", "presto", "status") != 0) {
+                LOGGER.error("Presto is not running");
+                return false;
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+        }
+        return true;
+    }
+
+    public enum StopType
+    {
+        GRACEFUL, TERMINATE, KILL;
+
+        public static StopType fromString(String name)
+        {
+            try {
+                return valueOf(name.toUpperCase());
+            }
+            catch (IllegalArgumentException e) {
+                return GRACEFUL;
```

Comment:
Maybe make this a constant in `AgentControlUtils`, so the default can be changed more easily.